app/plugin.go

```diff
@@ -662,16 +674,54 @@ func (a *App) installPrepackagedPlugin(pluginPaths *pluginSignaturePaths) (*plug
 	}
 	defer os.RemoveAll(tmpDir)
 
-	plug, pluginDir, err := getPrepackagedPlugin(pluginPaths, fileReader, tmpDir)
+	plugin, pluginDir, err := getPrepackagedPlugin(pluginPaths, fileReader, tmpDir)
 	if err != nil {
 		return nil, errors.Wrapf(err, "Failed to get prepackaged plugin %s", pluginPaths.path)
 	}
 
-	if _, err := a.installExtractedPlugin(plug.Manifest, pluginDir, true); err != nil {
-		return nil, errors.Wrapf(err, "Failed to install extracted prepackaged plugin %s", pluginPaths.path)
+	if !*a.Config().PluginSettings.AutomaticPrepackagedPlugins {
+		return plugin, nil
+	}
+
+	if state, ok := a.Config().PluginSettings.PluginStates[plugin.Manifest.Id]; ok && state.Enable {
+		newVersion, err := isNewVersion(plugin.Manifest, avaliablePlugins)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Failed to parse plugin version %s for prepackaged plugin %s", plugin.Manifest.Version, plugin.Manifest.Id)
```

Comment:
The message for this error implies too much knowledge about how `isNewVersion` works -- should it just say, `failed to check for an existing, newer version of %s`?